gattc_ninafw.go

```diff
@@ -0,0 +1,318 @@
+//go:build ninafw
+
+package bluetooth
+
+import "errors"
+
+var (
+	errNotYetImplemented         = errors.New("bluetooth: not yet implemented")
+	errNoWrite                   = errors.New("bluetooth: write not permitted")
+	errNoWriteWithoutResponse    = errors.New("bluetooth: write without response not permitted")
+	errWriteFailed               = errors.New("bluetooth: write failed")
+	errNoRead                    = errors.New("bluetooth: read not permitted")
+	errReadFailed                = errors.New("bluetooth: read failed")
+	errNoNotify                  = errors.New("bluetooth: notify/indicate not permitted")
+	errEnableNotificationsFailed = errors.New("bluetooth: enable notifications failed")
+	errServiceNotFound           = errors.New("bluetooth: service not found")
+	errCharacteristicNotFound    = errors.New("bluetooth: characteristic not found")
+)
+
+const (
+	maxDefaultServicesToDiscover        = 6
+	maxDefaultCharacteristicsToDiscover = 8
+)
+
+const (
+	charPropertyBroadcast            = 0x01
+	charPropertyRead                 = 0x02
+	charPropertyWriteWithoutResponse = 0x04
+	charPropertyWrite                = 0x08
+	charPropertyNotify               = 0x10
+	charPropertyIndicate             = 0x20
+)
+
+// DeviceService is a BLE service on a connected peripheral device.
+type DeviceService struct {
+	uuid UUID
+
+	device                 *Device
+	startHandle, endHandle uint16
+}
+
+// UUID returns the UUID for this DeviceService.
+func (s *DeviceService) UUID() UUID {
+	return s.uuid
+}
```

Comment:
Reviewing this code I realized that the pointer receiver here is not necessary. I believe the optimizer will optimize it away regardless, but using `s DeviceService` might be better. See: https://github.com/tinygo-org/bluetooth/pull/218

Same for `Discoverservices` below, and similar methods on `DeviceCharacteristic`.

Reply:
Tried to do this. Compiled but it did not work as expected.

Let's do this in a follow-up PR.